front/components/assistant/conversation/ActionValidationProvider.tsx

```diff
@@ -134,26 +141,37 @@ export function ActionValidationProvider({
 
   const handleSubmit = (approved: MCPValidationOutputType) => {
     void sendCurrentValidation(approved);
-    setIsProcessing(false);
+
     const foundItem = takeNextFromQueue();
-    if (!foundItem) {
+    if (foundItem) {
+      setCurrentValidation(foundItem);
+    } else {
+      // To avoid content flickering, we will clear out the current validation in onDialogAnimationEnd.
       setIsDialogOpen(false);
     }
   };
 
-  useEffect(() => {
-    if (currentValidation) {
-      setIsDialogOpen(true);
+  // To avoid content flickering, we will clear out the current validation when closing animation ends.
+  const onDialogAnimationEnd = () => {
+    // This is safe to check because the dialog closing animation is triggered after isDialogOpen is set to false.
+    if (!isDialogOpen) {
+      setCurrentValidation(null);
+      setErrorMessage(null);
```

Comment:
Right now we clear an error when modal is open, but I think it makes more sense to clear when we close it?